kernel/kernel-api/src/main/java/io/delta/kernel/internal/util/FileNames.java

```diff
@@ -136,18 +155,22 @@ public static boolean isClassicCheckpointFile(String fileName) {
     return CLASSIC_CHECKPOINT_FILE_PATTERN.matcher(fileName).matches();
   }
 
-  public static boolean isMulitPartCheckpointFile(String fileName) {
+  public static boolean isMultiPartCheckpointFile(String fileName) {
     return MULTI_PART_CHECKPOINT_FILE_PATTERN.matcher(fileName).matches();
   }
 
   public static boolean isV2CheckpointFile(String fileName) {
     return V2_CHECKPOINT_FILE_PATTERN.matcher(fileName).matches();
   }
 
-  public static boolean isCommitFile(String fileName) {
-    String filename = new Path(fileName).getName();
-    return DELTA_FILE_PATTERN.matcher(filename).matches()
-        || UUID_DELTA_FILE_REGEX.matcher(filename).matches();
+  public static boolean isCommitFile(String filePathStr) {
+    Path filePath = new Path(filePathStr);
+    String fileName = filePath.getName();
+    String fileParentName = filePath.getParent().getName();
```

Comment:
[nit] Can we fetch the fileParentName only when it is needed? These Path/URI operations are slow and may cause slowness when applied on large number of objects.

    if (DELTA_FILE_PATTERN.matcher(fileName).matches()) {
      return true
    } else {
      String fileParentName = filePath.getParent().getName();
      return COMMIT_SUBDIR.equals(fileParentName)
        && UUID_DELTA_FILE_REGEX.matcher(fileName).matches());
    }